manticore/ethereum/manticore.py

```
@@ -1129,75 +1144,34 @@ def _on_concretize(self, state, func, data, result):
-                data: is the symbolic value of the function domain
-                known_pairs: in/out is a dictionary containing known pairs from {domain, range}"""
-            # We are adding a new pair to the symbolic pairs
-            # Reset the soundcheck if True
-            if state.context.get("soundcheck", None) == True:
-                state.context["soundcheck"] = None
-
-            data_var = state.new_symbolic_buffer(len(data))  # FIXME: generalize to bitvec
-            state.constrain(data_var == data)
-            data = data_var
-            # symbolic_pairs is the pairs known locally for this symbolic function
-            symbolic_pairs = state.context.get(f"symbolic_func_sym_{name}", [])
-            # lets make a fresh 256 bit symbol representing any potential hash
-            value = state.new_symbolic_value(256)
-
-            for x, y in symbolic_pairs:
-                # if we found another pair that matches use that instead
-                # the duplicated pair is not added to symbolic_pairs
-                if state.must_be_true(Operators.OR(x == data, y == value)):
-                    constraint = Operators.AND(x == data, y == value)
-                    state.constrain(constraint)
-                    data, value = x, y
-                    break
-            else:
-                # bijectiveness; new pair is added to symbolic_pairs
-                for x, y in symbolic_pairs:
-                    if len(x) == len(data):
-                        constraint = (x == data) == (y == value)
-                    else:
-                        constraint = y != value
-                    state.constrain(constraint)  # bijective
-                symbolic_pairs.append((data, value))
-                state.context[f"symbolic_func_sym_{name}"] = symbolic_pairs
-
-        else:
+        value = None
+        try:
             value = func(data)
+        except Exception as e:
+            pass  # function is unknown
```

Is it always okay to silence the exception here? Are there cases where we might want at least a warning instead?

---

Maybe a better API would be to make func return None when it's not defined